Lib/test/test_clinic.py

```diff
@@ -1471,7 +1570,100 @@ def test_illegal_c_identifier(self):
             test.fn
                 a as 17a: int
         """
-        self.expect_failure(block, err)
+        self.expect_failure(block, err, lineno=2)
+
+    def test_cannot_convert_special_method(self):
+        err = "__len__ is a special method and cannot be converted"
+        block = """
+            class T "" ""
+            T.__len__
+        """
+        self.expect_failure(block, err, lineno=1)
+
+    def test_cannot_specify_pydefault_without_default(self):
+        err = "You can't specify py_default without specifying a default value!"
+        block = """
+            fn
+                a: object(py_default='NULL')
+        """
+        self.expect_failure(block, err, lineno=1)
+
+    def test_vararg_cannot_take_default_value(self):
+        err = "Vararg can't take a default value!"
+        block = """
+            fn
+                *args: object = None
+        """
+        self.expect_failure(block, err, lineno=1)
+
+    def test_invalid_legacy_converter(self):
+        err = "fhi is not a valid legacy converter"
+        block = """
+            fn
+                a: 'fhi'
+        """
+        self.expect_failure(block, err, lineno=1)
```

Comment:
Ideally the lineno here would be 2, right?

Reply:
Yes; I mentioned this in an earlier PR as well; for some of the tests (the ones using FakeClinic), the line numbers are off by one. I _suspect_ it is caused by FakeClinic, because on the command-line, clinic always reports accurate line numbers.

Reply:
I guessed this was the case; just wanted to check!

Reply:
Also, the line numbers in the ClinicWholeFile test case are _not_ off by one; they are accurate, as expected :)